### PR TITLE
[CS50] - Learning about tests in Python

### PR DESCRIPTION
**Pull request by inchoate**
This PR shows how basic tests work in Python.

To run:
```
python3 runTests.py
```

### CODE REVIEW

File: week2/runTests.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import unittest
+loader = unittest.TestLoader()
+start_dir = 'tests'
+suite = loader.discover(start_dir)
+
+runner = unittest.TextTestRunner()
+runner.run(suite)
```
Comment (inchoate) [inline week2/runTests.py:7]: I would write this following PEP8 standards:

```
import unittest

START_DIR = 'tests'

loader = unittest.TestLoader()
suite = loader.discover(START_DIR)
runner = unittest.TextTestRunner()

runner.run(suite)
```

In Python the module/file should be named `run_tests.py`.

File: week2/solutions/charIsIn.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+def isIn(char, aStr):
```
Comment (inchoate) [inline week2/solutions/charIsIn.py:1]: Python naming should be `snake_case`:
```
def is_in(char, a_str):
    ...
```

File: week2/solutions/evalQuadratic.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+def evalQuadratic(a, b, c, x):
```
Comment (inchoate) [inline week2/solutions/evalQuadratic.py:1]: Same here,
```
def eval_quadratic
```

File: week2/solutions/gcdIter.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+def gcdIter(a, b):
```
Comment (inchoate) [inline week2/solutions/gcdIter.py:1]: same

File: week2/solutions/gcdRecur.py [hunk 5]
```diff
@@ -0,0 +1,6 @@
+def gcdRecur(a, b):
```
Comment (inchoate) [inline week2/solutions/gcdRecur.py:1]: same

File: week2/solutions/guessTheNumber.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
+import random
+
+def getUserInput(guess):
```
Comment (inchoate) [inline week2/solutions/guessTheNumber.py:3]: `get_user_input`

File: week2/solutions/guessTheNumber.py [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  + " Enter 'l' to indicate the guess is too low."
+  + " Enter 'c' to indicate I guessed correctly.")) 
+
+userInput = ''
+guess = 50 #random.randint(0, 100)
+high = 100
+low = 0
+
+print('Please think of a number between 0 and 100!')
+while userInput != 'c':
```
Comment (inchoate) [inline week2/solutions/guessTheNumber.py:15]: `user_input`

File: week2/tests/test_charIsIn.py [hunk 8]
```diff
@@ -0,0 +1,32 @@
+import unittest
+import os
+import sys
+import solutions.charIsIn as charIsIn
+
```
Comment (inchoate) [inline week2/tests/test_charIsIn.py:5]: Imports should always be lexically sorted, and in 3 sections:
(1) System libraries + a newline
(2) 3rd party libraries + a newline
(3) Your project's imports + a newline

File: week2/tests/test_charIsIn.py [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+class isInTest1(unittest.TestCase):
+  def test(self):
+    self.assertEqual(charIsIn.isIn('z', 'abz'), True)
+
+class isInTest2(unittest.TestCase):
```
Comment (inchoate) [inline week2/tests/test_charIsIn.py:10]: Class naming in Python is capital snake case:
```
IsInTest2
```

File: week2/tests/test_checkOdd.py [hunk 10]
```diff
@@ -0,0 +1,14 @@
+import unittest
+import os
+import sys
```
Comment (inchoate) [inline week2/tests/test_checkOdd.py:3]: Sort imports

**Review by inchoate — COMMENTED** [review]
Some stylistic Python changes requested.

**Review by kayleabox — APPROVED** [review]

**Pull request merged**